### PR TITLE
Add `ExecConfig::drop_env` to filter host evars for auth providers

### PR DESCRIPTION
**Pull request by aviramha**
Solves #1060

### CODE REVIEW

File: kube-client/src/config/file_config.rs [hunk 1]
```diff
@@ -243,6 +243,11 @@ pub struct ExecConfig {
     /// TODO: These are unioned with the host's environment, as well as variables client-go uses to pass argument to the plugin.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub env: Option<Vec<HashMap<String, String>>>,
+    /// drop_env specifies which environment arguments of the host process shouldn't
+    /// pass to the auth plugin.
+    /// This doesn't exist upstream (yet?)
```
Comment (clux) [inline kube-client/src/config/file_config.rs:248]:
```suggestion
    /// Specifies which environment variables the host should avoid passing to the auth plugin.
    ///
    /// This does currently not exist upstream and cannot be specified on disk.
    /// It has been suggested in client-go via https://github.com/kubernetes/client-go/issues/1177
```

**Comment by aviramha** [comment]
Suggested change committed.

**Review by clux — APPROVED** [review]

**Comment by aviramha** [comment]
Anything needed from my side atm?

**Comment by clux** [comment]
Nah, all good, I was just giving a little time to see if anyone else had input. Happy to merge this, and will if no one else comments upstream or here for the next couple of days.

**Comment by teozkr** [comment]
:+1: from me as long as it's `#[serde(skip)]` until there's upstream consensus.

**Pull request merged**